pymobiledevice3/services/dvt/testmanaged/xcuitest.py

```diff
@@ -0,0 +1,259 @@
+import logging
+import time
+from typing import Optional
+
+from bpylist2 import archiver
+from packaging.version import Version
+
+from pymobiledevice3.lockdown_service_provider import LockdownServiceProvider
+from pymobiledevice3.services.afc import AfcService
+from pymobiledevice3.services.dvt.dvt_secure_socket_proxy import DvtSecureSocketProxyService
+from pymobiledevice3.services.dvt.dvt_testmanaged_proxy import DvtTestmanagedProxyService
+from pymobiledevice3.services.dvt.instruments.process_control import ProcessControl
+from pymobiledevice3.services.remote_server import (
+    NSURL, NSUUID, Channel, ChannelFragmenter, MessageAux, XCTestConfiguration,
+    dtx_message_header_struct, dtx_message_payload_header_struct)
+
+logger = logging.getLogger(__name__)
+
+
+class XCUITestService:
+    IDENTIFIER = 'dtxproxy:XCTestManager_IDEInterface:XCTestManager_DaemonConnectionInterface'
+    XCODE_VERSION = 36 # not important
+
+    def __init__(self,
+                 service_provider: LockdownServiceProvider,
+                 afc: AfcService,
+                 app_info: dict):
+        self.service_provider = service_provider
+        self.afc = afc
+        self.pctl = self.init_process_control()
+        self.app_info = app_info
+        self.product_major_version = Version(service_provider.product_version).major
+
+    def run(self, 
+            bundle_id: str,            
+            test_runner_env: Optional[dict] = None, 
```

Comment:
dict and list typings are only supported on python3.9+
this project supports python3.8 since it hasn't reached its end of line

Reply:
Just test, it works on python 3.8.0

```python
from typing import Optional

def foo(a: Optional[dict] = None, b: list = None):
    print(a, b)

foo({}, [])
```